Mathlib/FieldTheory/JacobsonNoether.lean

```diff
@@ -0,0 +1,209 @@
+/-
+Copyright (c) 2024 F. Nuccio, H. Zheng, W. He, S. Wu, Y. Yuan, W. Jiao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Filippo A. E. Nuccio, Huanyu Zheng, Sihan Wu, Wanyi He, Weichen Jiao, Yi Yuan
+-/
+import Mathlib.Algebra.CharP.LinearMaps
+import Mathlib.Algebra.Field.Power
+import Mathlib.FieldTheory.PurelyInseparable
+
+/-!
+# The Jacobson-Noether theorem
+
+This file contains a proof of the Jacobson-Noether theorem and some auxiliary lemmas.
+Here we discuss different cases of characteristics of
+the noncommutative division algebra `D` with the center `k`.
+
+## Main Results
+
+- `Jacobson_Noether` : Let `D` be a noncommutative division algebra with the center `k`.
+  If `D` is algebraic over `k`,
+  then there exists an element of `D \ k` which is separable over `k`.
+
+## Notations
+
+- `D` is a noncommutative division algebra
+- `k` is the center of `D`
+
+## Implementation Notes
+
+## Reference
+* <https://ysharifi.wordpress.com/2011/09/30/the-jacobson-noether-theorem/>
+-/
+
+namespace JacobsonNoether
+
+variable {D : Type*} [DivisionRing D]
+
+local notation3 "k" => (Subring.center D)
+local notation3 "f" => (LinearMap.mulLeft k)
```

Comment:
I would suggest that you remove `f` and `g` altogether. They already have a name in mathlib and if you `open LinearMap` this name is not very long), and their commutativity is also in the library, so you should rather use it directly rather than creating a new, ad-hoc lemma. You can keep `δ` and the related business, nevertheless (marking them `private` is certainly a good idea).